### PR TITLE
dropped quay.io from docker paths

### PR DESCRIPTION
**Pull request by mattheww95**
Removed docker `quay.io` tags.

### CODE REVIEW

**Comment by sgsutcliffe** [comment]
Included the override container registry process (for more information see [phac-nml pipeline standards](https://github.com/phac-nml/pipeline-standards?tab=readme-ov-file#522-configuring-module-software-with-private-or-alternate-container-registries)) to allow the container directives to have a conditional check in the nested ternary operator. Added this to the `pilon_interactive` process used in the module pilon_polisher.nf` which uses docker.hub rather than the default `quay.io`.

**Comment by sgsutcliffe** [comment]
Added the default container registries to `quay.io` 697cbb2

File: nextflow.config [hunk 1]
```diff
@@ -173,7 +173,7 @@ params {
 
     abricate {
         singularity = "https://depot.galaxyproject.org/singularity/abricate%3A1.0.1--ha8f3691_1"
-        docker = "quay.io/biocontainers/abricate:1.0.1--ha8f3691_1"
+        docker = "biocontainers/abricate:1.0.1--ha8f3691_1"
```
Comment (sgsutcliffe) [inline nextflow.config:176]: If we remove the `quay.io` registry prefix we need to include the `docker.registry = 'quay.io'` 697cbb2

**Comment by sgsutcliffe** [comment]
Resolving the branch conflict included updating the `locidex` container to the dockerhub repository. I'll add the same conditional operator here as well.

File: nextflow.config [hunk 2]
```diff
@@ -1041,6 +1043,17 @@ profiles {
...
+// Set default registry for Apptainer, Docker, Podman and Singularity independent of -profile
+// Will not be used unless Apptainer / Docker / Podman / Singularity are enabled
+// Set to your registry if you have a mirror of containers
+apptainer.registry   = 'quay.io'
+docker.registry      = 'quay.io'
+podman.registry      = 'quay.io'
+singularity.registry = 'quay.io'
+
+// Override the default Docker registry when required
+process.ext.override_configured_container_registry = false
```
Comment (apetkau) [inline nextflow.config:1055]: By default this should be `true` instead of `false`.

**Review by apetkau — CHANGES_REQUESTED** [review]
This looks great. Thanks so much for your work on this Steven and making all those changes/testing things out 😄 

I just have one comment.

File: nextflow.config (hunk 2 above)
Comment (sgsutcliffe) [reply nextflow.config:1055]: Yes! Must have slipped back in from testing. Good catch!

**Comment by mattheww95** [comment]
Just looked everything over! It looks great and thank you so much for your help!

**Comment by sgsutcliffe** [comment]
In addition to updating containers to use a default registry `quay.io` we will replace the python container.
```bash
    python3 {
        singularity = "biocontainers/staramr:0.10.0--pyhdfd78af_0"
        docker = "biocontainers/staramr:0.10.0--pyhdfd78af_0"
    }
```

Which has Python 3.11.4 running under the hood. So will replace with Python 3.12 (the closest quay.io container)

**Review by apetkau — APPROVED** [review]
Looks good. Thanks so much Steven 😄

**Pull request merged**